src/id_token_provider.erl

```diff
@@ -68,19 +70,25 @@ handle_info({refresh, Provider}, State) ->
   #{exp_at := ExpAt} = refresh(Provider),
   Now = id_token_util:now_gregorian_seconds(),
   RevalidateTime = ExpAt - ?REVALIDATE_DELAY,
-  case Now < RevalidateTime of
-    true ->
-      timer:send_after((RevalidateTime - Now) * 1000,
-                       self(), {refresh, Provider});
-    false ->
-      %% Not enough time for revalidation,
-      %% let the first request pay the price
-      ok
-  end,
+  Delay =
+    case Now < RevalidateTime of
+      true ->
+        (RevalidateTime - Now) * 1000;
+      false ->
+        %% Not enough time for revalidation, let the first request pay
+        %% the price and re-initiate async_revalidate-loop after 60 seconds
+        60_000
+    end,
+  timer:send_after(Delay, self(), {refresh, Provider}),
   {noreply, State}.
 
-maybe_refresh(Provider, #{force_refresh := true}) ->
```

Comment:
Probably good to keep this one as well to have a way to refresh the cache in case of issues.